### DISCUSSION TITLE
ssh-grunt: Get public keys from alternative source

### DISCUSSION THREAD

**Discussion by iantybo** [gruntwork-io/knowledge-base#536] [Help] [answered]
I am using Okta and would like to have ssh-grunt pull public keys from there so I can use OpenVPN, is this possible?

---

<ins datetime="2022-08-17T13:49:00Z">
  <p><a href="https://support.gruntwork.io/hc/requests/109141">Tracked in ticket #109141</a></p>
</ins>

**Comment by yorinasub17** [comment, marked as answer]
`ssh-grunt` currently can not pull SSH keys or users from sources other than IAM. I filed https://github.com/gruntwork-io/terraform-aws-security/issues/709 to track this request.

Similarly, `openvpn-admin` (the utility for managing and provisioning users) currently can not manage users outside of the internal s3 backed certificate store. Due to the architecture of OpenVPN, this is harder to implement and we are unlikely to roll out a feature that makes this possible. We recommend using an alternative VPN system that supports SSO, such as Tailscale.
